pkg/aws/core.go

```diff
@@ -559,28 +559,41 @@ func (d *Driver) ListMachines(ctx context.Context, req *driver.ListMachinesReque
 		},
 	}
 
-	runResult, err := client.DescribeInstances(ctx, input)
-	if err != nil {
-		klog.Errorf("AWS plugin is returning error while describe instances request is sent: %s", err)
-		return nil, status.Error(codes.Internal, err.Error())
-	}
-
 	listOfVMs := make(map[string]string)
-	for _, reservation := range runResult.Reservations {
-		for _, instance := range reservation.Instances {
-
-			machineName := ""
-			for _, tag := range instance.Tags {
-				if *tag.Key == "Name" {
-					machineName = *tag.Value
-					break
+	var nextToken *string
+	pageCount := 0
+
+	for {
+		input.NextToken = nextToken
+
+		runResult, err := client.DescribeInstances(ctx, input)
+		if err != nil {
+			klog.Errorf("AWS plugin is returning error while describe instances request is sent: %s (NextToken: %s)", err, ptr.Deref(nextToken, "<nil>"))
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		pageCount++
+
+		for _, reservation := range runResult.Reservations {
+			for _, instance := range reservation.Instances {
+				machineName := ""
+				for _, tag := range instance.Tags {
+					if *tag.Key == "Name" {
+						machineName = *tag.Value
+						break
+					}
 				}
+				listOfVMs[encodeInstanceID(providerSpec.Region, *instance.InstanceId)] = machineName
 			}
-			listOfVMs[encodeInstanceID(providerSpec.Region, *instance.InstanceId)] = machineName
 		}
+
+		if runResult.NextToken == nil {
```

Comment:
```suggestion
		if runResult.NextToken == nil || *runResult.NextToken == "" {
```
There was an additional check added for `azure`, is there a reason why its not necessary here?